### ISSUE TITLE
Latest release bricks my Pi

### ISSUE THREAD

**Issue by spenadel** [taoteh1221/Slideshow_Crypto_Ticker#46]
Hi - I love the ticker.  My pi is on Bullseye.  When I upgraded yesterday, it trashed my installation and I had to re-image my pi.  Have you seen this behavior?  I thought I had a full backup image, but apparently don't, so I need to install an earlier version.  Do you have a suggestion as to which version I should downgrade to?

Thanks

**Comment by taoteh1221** [comment]
Hello @spenadel sorry to hear this and thanks for the report. What disk image are you using (raspi os, armbian, dietpi, etc), and what pi model? Try a couple versions earlier. I'll attempt to replicate the issue this weekend on my pi3.

**Comment by spenadel** [comment]
This is a pi4 and I would have put raspi on it.  Thanks for looking into this.  When I tried it again an hour ago, the raspi would not display the desktop on the tft, nor could I ssh or VNC into it, but I could ping it.  Does the latest version upgrade the OS, or perhaps reset the ssh and vnc options on the pi?  That could explain what I saw.

**Comment by taoteh1221** [comment]
Ok, thanks for the details. I have a pi4 so I'll use that for testing. The latest releases change to using LXDE desktop's autostart feature, instead of a systemd service (this runs bash scripts more reliably for the recent multi-web-browser support feature). There are other changes related to lightdm auto login too, BUT that is only if raspberry pi os is not detected (for armbian etc, as raspi os has auto login on by default).

**Comment by taoteh1221** [comment]
One other thing @spenadel : Are you using 32bit or 64bit disk image? Raspi os now has 64bit disk images, BUT they only just started doing this...may be bugs in 64bit for a few months, unrelated to the ticker app install.

**Comment by taoteh1221** [comment]
That's odd if you're saying that SSH that was already installed by you previously is acting up, unless you have a firewall on there not properly configured. I'm not sure off the top of my head how the ticker could cause this, but I'll check out the auto-installers logic this Saturday. Sorry about this.

**Comment by spenadel** [comment]
If I recall, it was 64-bit.  I want to ensure that the base Raspi image is not the issue, because I went back to 5.9 and had a similar issue.  So, I’m installing Raspi 64-bit and starting from scratch.  I’ll build a new base image and start with 5.14 and see what happened.  BTW, what drove me to this point was trying to log into the Admin page of the portfolio tracker and not being able to sign in.  There was no file within /cache directory tree to delete, and I upon reinstallation, I was not asked to set an admin password.  Will advise on my progress today.

From: taoteh1221 ***@***.***>
Date: Thursday, April 21, 2022 at 8:09 AM
To: taoteh1221/Slideshow_Crypto_Ticker ***@***.***>
Cc: Lee Spenadel ***@***.***>, Mention ***@***.***>
Subject: Re: [taoteh1221/Slideshow_Crypto_Ticker] Latest release bricks my Pi (Issue #46)

One other thing @spenadel<https://github.com/spenadel> : Are you using 32bit or 64bit disk image? Raspi os now has 64bit disk images, BUT they only just started doing this...may be bugs in 64bit for a few months, unrelated to the ticker app install.

**Comment by taoteh1221** [comment]
Ok, thanks for all the feedback. I freed up some time this morning to look into all this. BTW, the main issue with the ticker ever affecting anything would be it's auto-installation script, which sets up starting at boot etc. The ticker itself is just an index.html file opened as a local file by a web browser...not much there that would affect an entire system's stability. If you still have any issues with the portfolio tracker, feel free to open an issue in that repo here and we'll get that fixed too (sounds like it may be a php8 compatibility issue): https://github.com/taoteh1221/Open_Crypto_Tracker/issues

**Comment by taoteh1221** [comment]
@spenadel I tested the latest (2022-4-4) 32bit "Raspberry Pi OS with desktop and recommended software" disk image on my pi3 (I won't have access to my pi4 until saturday), and encounter no errors using the ticker auto-installer. ![Screenshot from 2022-04-21 11-52-33](https://user-images.githubusercontent.com/1760489/164501257-f5c580be-ebb1-4a66-a8b1-5f4ae70d14ac.png)

Run this on the above-mentioned disk image, it uses the latest auto-installation code:
```
wget --no-cache -O TICKER-INSTALL.bash https://git.io/Jqzjk;chmod +x TICKER-INSTALL.bash;sudo ./TICKER-INSTALL.bash
```

There is the "edge case" possibility that if you ran the v3.09.4 upgrade a few weeks ago, that could 'brick' your desktop autostart file, as there was a bug I fixed for that 2 weeks ago in v3.09.5: 

https://github.com/taoteh1221/Slideshow_Crypto_Ticker/commit/ccec436b4831d28f3f03ff05b2be54c35f596ab0

As for replicating the portfolio tracker bug, all I've found so far is a small issue with fresh installs initially showing a blank screen first run (then shows the admin registration page after reloading the app). In config.php at the very top you can enable debugging output to apache's error logs, and open a new issue on the portfolio tracker github repo / paste the log output into the description, to help me debug any continuing issues with that: 

https://github.com/taoteh1221/Open_Crypto_Tracker/blob/main/config.php#L18

**Comment by taoteh1221** [comment]
@spenadel Just for clarification that I understand correctly: You are running the LCD screen ticker, AND the web-app portfolio tracker on the same machine, right? That should be fine on a pi3 and higher, so you should be all set on a pi4. 

I saw you quoting version numbers for the portfolio tracker here (5.X), while the LCD screen ticker app is only up to v3.X as of now. Just want to make sure I understand which app you are talking about when trying to replicate issues you are having.

**Comment by taoteh1221** [comment]
My reasoning behind having you test with 32bit disk images is that the raspberry pi team has only just started making 64bit images 2 months ago for the first time EVER, so their could be bugs in those packages they may need to work out:

https://www.tomshardware.com/news/raspberry-pi-64-bit-os-released

So we want to rule that out as a possibility. Hope this makes sense.

**Comment by taoteh1221** [comment]
If you read down a bit in that article ( at: https://www.tomshardware.com/news/raspberry-pi-64-bit-os-released )

> We asked Hollingworth to explain the current status of Raspberry Pi OS 64-bit and in his email Hollingworth explains that the 64-bit release has "had less test time than the 32-bit build and therefore might have issues we've not found yet. But otherwise it has equal functionality where it is under our control to provide it." This means that for the majority of users, the move will be friction-less. But of course there will be a few teething issues, most likely from specific edge cases. If you find an issue, report it!

**Comment by spenadel** [comment]
I ran both apps on the same Pi4.  Your version comment made me realize that I may have been installing the incorrect software (portfolio) versus backing down on the screen ticker code.

From: taoteh1221 ***@***.***>
Date: Thursday, April 21, 2022 at 12:30 PM
To: taoteh1221/Slideshow_Crypto_Ticker ***@***.***>
Cc: Lee Spenadel ***@***.***>, Mention ***@***.***>
Subject: Re: [taoteh1221/Slideshow_Crypto_Ticker] Latest release bricks my Pi (Issue #46)

@spenadel<https://github.com/spenadel> Just for clarification that I understand correctly: You are running the LCD screen ticker, AND the web-app portfolio tracker on the same machine, right? That should be fine on a pi3 and higher, so you should be all set on a pi4.

I saw you quoting version numbers for the portfolio tracker here (5.X), while the LCD screen ticker app is only up to v3.X as of now. Just want to make sure I understand which app you are talking about when trying to replicate issues you are having.

**Comment by taoteh1221** [comment]
> I ran both apps on the same Pi4. Your version comment made me realize that I may have been installing the incorrect software (portfolio) versus backing down on the screen ticker code. 

No worries, we'll figure it out. Keep in mind v5.12.2 and higher (of portfolio app) renames the admin config array (so the old config file is incompatible...old configs are auto-backup 'ed in the same directory during upgrades). Sorry about this, some of the final growing pains optimizing it.

I recommend trying the latest versions of everything on the 32bit disk image. I just did this on a pi3 this morning, with no issues other than this fairly minor one: 

https://github.com/taoteh1221/Open_Crypto_Tracker/issues/41

**Comment by taoteh1221** [comment]
FYI @spenadel if you still get login errors on the portfolio tracker, after enabling debugging at top of config.php like I mentioned, run this command in a terminal to see apache error logs in real time:

```
sudo tail -f /var/log/apache2/error.log
```

**Comment by spenadel** [comment]
I’ve just reimaged my SD with the 32-bit version of raspi.

I reinstalled the crypto-ticker and it’s running as expected.  I did get the page to create an admin account, but no matter how many times I input an admin username/password and image text, the page refreshes with no saved data, so I’m stuck at this juncture trying to setup the portfolio tracker.

Thanks
Lee

From: taoteh1221 ***@***.***>
Date: Thursday, April 21, 2022 at 2:23 PM
To: taoteh1221/Slideshow_Crypto_Ticker ***@***.***>
Cc: Lee Spenadel ***@***.***>, Mention ***@***.***>
Subject: Re: [taoteh1221/Slideshow_Crypto_Ticker] Latest release bricks my Pi (Issue #46)

I ran both apps on the same Pi4. Your version comment made me realize that I may have been installing the incorrect software (portfolio) versus backing down on the screen ticker code.

No worries, we'll figure it out. Keep in mind v5.12.2 and higher (of portfolio app) renames the admin config array (so the old config file is incompatible...old configs are auto-backup 'ed in the same directory during upgrades). Sorry about this, some of the final growing pains optimizing it.

I recommend trying the latest versions of everything on the 32bit disk image. I just did this on a pi3 this morning, with no issues other than this fairly minor one:

taoteh1221/Open_Crypto_Tracker#41<https://github.com/taoteh1221/Open_Crypto_Tracker/issues/41>

**Comment by spenadel** [comment]
I was able to create the admin account using Safari.  There could be an add-on in Firefox that’s creating the problem, which I will address.  Thank you for your support – I’m good now.

From: taoteh1221 ***@***.***>
Date: Thursday, April 21, 2022 at 4:01 PM
To: taoteh1221/Slideshow_Crypto_Ticker ***@***.***>
Cc: Lee Spenadel ***@***.***>, Mention ***@***.***>
Subject: Re: [taoteh1221/Slideshow_Crypto_Ticker] Latest release bricks my Pi (Issue #46)

FYI @spenadel<https://github.com/spenadel> if you still get login errors on the portfolio tracker, after enabling debugging at top of config.php like I mentioned, run this command in a terminal to see apache error logs in real time:

sudo tail -f /var/log/apache2/error.log

**Comment by taoteh1221** [comment]
> I was able to create the admin account using Safari. There could be an add-on in Firefox that’s creating the problem, which I will address. Thank you for your support – I’m good now. 

Glad you found a workaround. I'll test admin registration with firefox, and fix any issues I notice. I usually use chrome, so I haven't yet done this:

https://github.com/taoteh1221/Open_Crypto_Tracker/issues/42

Thanks for all the feedback Lee, I appreciate it. :+1:

**Comment by spenadel** [comment]
Why would I be seeing this log entry:

[Thu Apr 21 16:13:37.250907 2022] [access_compat:error] [pid 620:tid 2852045824] [client 166.77.201.16:48634] AH01797: client denied by server configuration: /var/www/html/plugins/htaccess_security_check.dat



From: taoteh1221 ***@***.***>
Date: Thursday, April 21, 2022 at 4:42 PM
To: taoteh1221/Slideshow_Crypto_Ticker ***@***.***>
Cc: Lee Spenadel ***@***.***>, Mention ***@***.***>
Subject: Re: [taoteh1221/Slideshow_Crypto_Ticker] Latest release bricks my Pi (Issue #46)

I was able to create the admin account using Safari. There could be an add-on in Firefox that’s creating the problem, which I will address. Thank you for your support – I’m good now.

Glad you found a workaround. I'll test admin registration with firefox, and fix any issues I notice. I usually use chrome, so I haven't yet done this:

taoteh1221/Open_Crypto_Tracker#42<https://github.com/taoteh1221/Open_Crypto_Tracker/issues/42>

Thanks for all the feedback Lee, I appreciate it. 👍

**Comment by taoteh1221** [comment]
> Why would I be seeing this log entry: [Thu Apr 21 16:13:37.250907 2022] [access_compat:error] [pid 620:tid 2852045824] [client 166.77.201.16:48634] AH01797: client denied by server configuration: /var/www/html/plugins/htaccess_security_check.dat

That's the app on occasion checking that you have the proper htaccess security settings enabled on the web server. If it was ever NOT denied access to the cache directory via https, then the app would warn you in the interface that you needed to correct it. That's GOOD it's being denied access during it's self-checks. It means your cache directory is NOT accessible via a web address call. It's secure in that way.

**Comment by taoteh1221** [comment]
If you visit https://your.app.install.address/cache you'll see what I mean. You should be denied access.

**Comment by taoteh1221** [comment]
Actually in that particular case: https://your.app.install.address/plugins

That's checked as well for security on occasion.

**Comment by taoteh1221** [comment]
That's the apache error logs, correct?

**Comment by spenadel** [comment]
Unrelated question.  Have you found a way to programmatically turn off the display?  I want to turn it off at night, back on in the AM via cron job.  I’m using LCD-Show and have not found a way to manage this.  Any suggestions?

Thanks

From: taoteh1221 ***@***.***>
Date: Thursday, April 21, 2022 at 7:11 PM
To: taoteh1221/Slideshow_Crypto_Ticker ***@***.***>
Cc: Lee Spenadel ***@***.***>, Mention ***@***.***>
Subject: Re: [taoteh1221/Slideshow_Crypto_Ticker] Latest release bricks my Pi (Issue #46)

That's the apache error logs, correct?

**Comment by taoteh1221** [comment]
Sorry, I have no idea how to turn off the display other than MAYBE change the ticker to not keep the screen on during a 'do-not-disturb' time. Currently it virtually taps the Ctrl key every 60 seconds via a cron job.

**Comment by taoteh1221** [comment]
You could experiment if you can find this file, then disable the last line by putting a # in front of it:

https://github.com/taoteh1221/Slideshow_Crypto_Ticker/blob/main/bash/cron/keep-screensaver-off.bash#L11

That will disable the ticker from keeping the screen turned on all the time. It really all depends if your screen supports turning off when the screen goes blank, thats a hardware setting as far as I'm aware (not a software setting).

**Comment by taoteh1221** [comment]
@spenadel Thank you for the report on firefox not liking the portfolio tracker admin registration page. I have confirmed this bug, and will have a fix in the next release (v5.14.4): https://github.com/taoteh1221/Open_Crypto_Tracker/issues/42

**Comment by taoteh1221** [comment]
@spenadel The upcoming release of the tracker app (v5.15.4 will be out in a few days) has a fix completed that lets you register / reset the admin account using firefox: https://github.com/taoteh1221/Open_Crypto_Tracker/commit/a18b00ceccfec575fce66e76bea3e2c1ca0cee1f

Thanks very much for the feedback! :)
